### PR TITLE
Fix UDFs to be able to access nested objects

### LINKED ISSUE #13386: Allow UDFs to access nested object properties.
### Problem Statement

It seems like CrateDB currently only supports access to top-level properties for `OBJECT`s. Trying to access a nested property, returns a `NULL` value:

```sql
CREATE FUNCTION nested (obj OBJECT) 
    RETURNS STRING 
    LANGUAGE JAVASCRIPT 
    AS 'function nested(obj){ 
        return obj["l1"]; 
    }';                                                                                                                                                              

SELECT nested({l1='Hello'});                                                                                                                                     
/*
+---------+
| 'Hello' |
+---------+
| Hello   |
+---------+
SELECT 1 row in set (0.061 sec)
*/
```

```sql
CREATE FUNCTION nested2 (obj OBJECT) 
    RETURNS STRING 
    LANGUAGE JAVASCRIPT 
    AS 'function nested2(obj){ 
        return obj["l1"]["l2"]; 
    }';                                                                                                                                                              

SELECT nested2({l1={l2='Hello2'}});                                                                                                                              
/*
+------+
| NULL |
+------+
| NULL |
+------+
SELECT 1 row in set (0.094 sec)
*/
```

### Possible Solutions

allow access to nested properties in JS UDF

### Considered Alternatives

parse `OBJECT` to `TEXT` and use `JSON.parse()` in the `UDF`

```sql
cr> CREATE OR REPLACE FUNCTION nested3 (obj_text TEXT) 
    RETURNS STRING 
    LANGUAGE JAVASCRIPT 
    AS 'function nested3(obj_text){ 
        let obj = JSON.parse(obj_text); 
        return obj["l1"]["l2"]; 
    }';                                                                                                                                                              
CREATE OK, 1 row affected  (0.034 sec)
cr> SELECT nested3({l1={l2='Hello2'}}::TEXT);                                                                                                                        
+----------+
| 'Hello2' |
+----------+
| Hello2   |
+----------+
```

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes #13386 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: extensions/lang-js/src/test/java/io/crate/operation/language/JavaScriptUDFIntegrationTest.java [hunk 1]
```diff
@@ -108,4 +108,19 @@ public void test_udf_that_requires_array_arg_can_be_used_as_generated_column() t
         execute("select x from tbl");
         assertThat(TestingHelpers.printedTable(response.rows())).isEqualTo("27.4\n");
     }
+
+    // This tracks a bug: https://github.com/crate/crate/issues/13386
```
Comment (matriv) [inline extensions/lang-js/src/test/java/io/crate/operation/language/JavaScriptUDFIntegrationTest.java:112]: imho this is not necessary as it's not testing any complex code path/edge case.

**Review by matriv — APPROVED** [review]
LGTM, minor comment for the comment on the test.

**Review by mfussenegger — APPROVED** [review]
